cmd/fleet/cron.go

```diff
@@ -835,6 +835,19 @@ func newCleanupsAndAggregationSchedule(
 		schedule.WithJob("cleanup_unused_script_contents", func(ctx context.Context) error {
 			return ds.CleanupUnusedScriptContents(ctx)
 		}),
+		schedule.WithJob("cleanup_activities", func(ctx context.Context) error {
+			appConfig, err := ds.AppConfig(ctx)
+			if err != nil {
+				return err
+			}
+			if !appConfig.ActivityExpirySettings.ActivityExpiryEnabled {
+				return nil
+			}
+			// A maxCount of 5,000 means that the cron job will keep the activities (and associated tables)
+			// sizes in control for deployments that generate (5k x 24 hours) ~120,000 activities per day.
+			const maxCount = 5000
```

Comment:
It was not. Thanks for checking.

@noahtalerman Let me know if this sounds good.

Am adding a max to not lock the database for longs periods of time.

Reply:
@lucasmrod how does this affect the Fleet user's UX?

Does this mean that Fleet will delete a maximum of 120k activities each time the cleanup job runs? (once per day)

So, if a Fleet instance generates more than 120k activities per day, there will be some activities that are never removed?

Reply:
> Does this mean that Fleet will delete a maximum of 120k activities each time the cleanup job runs? (once per day)

5000 each time the cron runs, and it runs hourly (so ~120k every day)

> So, if a Fleet instance generates more than 120k activities per day, there will be some activities that are never removed?

Correct, it will never catch up and the storage will continue to grow (slower though).

Reply:
> 5000 each time the cron runs, and it runs hourly (so ~120k every day)

I think this limit makes sense as long as we can tweak it later if we learn it needs to be higher.

FYI @rachaelshaw

Reply:
Currently hardcoded to 5000, can be increased/made-configurable in the future for sure.